app/routers/invitation.py

```diff
@@ -0,0 +1,68 @@
+from typing import List, Union
+
+from fastapi import APIRouter, Depends, Request
+from fastapi.responses import RedirectResponse
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from starlette.status import HTTP_302_FOUND
+from starlette.templating import Jinja2Templates
+
+from app.database.database import get_db
+from app.database.models import Invitation
+from app.routers.share import accept
+
+
+templates = Jinja2Templates(directory="app/templates")
+
+router = APIRouter(
+    prefix="/invitations",
+    tags=["invitation"],
+    dependencies=[Depends(get_db)]
+)
+
+
+@router.get("/")
+def view_invitations(request: Request, db: Session = Depends(get_db)):
+    return templates.TemplateResponse("invitations.html", {
+        "request": request,
+        # TODO: create current user
+        # recipient_id should be the current user
+        # but because we don't have one yet,
+        # "get_all_invitations" returns all invitations
+        "invitations": get_all_invitations(session=db),
+    })
+
+
+@router.post("/")
+async def accept_invitations(
+    request: Request,
+    db: Session = Depends(get_db)
+):
+    data = await request.form()
+    invite_id = list(data.values())[0]
```

Comment:
next(data.values()) might work and save lot of space

Reply:
this probably won't work, data.values() isn't a iterator, it is an ValuesView type